frameworks/hdfs/tests/test_shakedown.py

```diff
@@ -250,8 +251,15 @@ def test_bump_data_nodes():
 
     sdk_marathon.bump_task_count_config(FOLDERED_SERVICE_NAME, 'DATA_COUNT')
 
-    check_healthy(count=DEFAULT_TASK_COUNT + 1)
-    sdk_tasks.check_tasks_not_updated(FOLDERED_SERVICE_NAME, 'data', data_ids)
+    try:
+        check_healthy(count=DEFAULT_TASK_COUNT + 1)
+        sdk_tasks.check_tasks_not_updated(FOLDERED_SERVICE_NAME, 'data', data_ids)
+
+    finally:
+        # TODO(elezar, nima) If one of the operations above (after the initial bump) fail, then
+        # the test will be terminated, and the number of tasks will remain at DEFAULT_TASK_COUNT + 1
+        sdk_marathon.bump_task_count_config(FOLDERED_SERVICE_NAME, 'DATA_COUNT', delta=-1)
```

Comment:
Thanks @NimaVaziri. That makes sense now. Note that subsequent health checks should then probably check for the new number of expected tasks. Would it make sense to update `DEFAULT_TASK_COUNT` (or have an `EXPECTED_TASK_COUNT` variable) to ensure that this is the case for subsequent tests?